p2p/trust/trustmetric.go

```diff
@@ -0,0 +1,394 @@
+package trust
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"math"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+var (
+	store *trustMetricStore
+)
+
+type peerMetricRequest struct {
+	Key  string
+	Resp chan *TrustMetric
+}
+
+type trustMetricStore struct {
+	PeerMetrics map[string]*TrustMetric
+	Requests    chan *peerMetricRequest
+	Disconn     chan string
+}
+
+func init() {
+	store = &trustMetricStore{
+		PeerMetrics: make(map[string]*TrustMetric),
+		Requests:    make(chan *peerMetricRequest, 10),
+		Disconn:     make(chan string, 10),
+	}
+
+	go store.processRequests()
+}
+
+type peerHistory struct {
+	NumIntervals int       `json:"intervals"`
+	History      []float64 `json:"history"`
+}
+
+func loadSaveFromFile(key string, isLoad bool, data *peerHistory) *peerHistory {
+	tmhome, ok := os.LookupEnv("TMHOME")
+	if !ok {
+		return nil
+	}
+
+	filename := filepath.Join(tmhome, "trust_history.json")
```

Comment:
cool. We should better integrate this with the rest of the package and use a new config.TrustHistory field for this instead of new hard_coded values. Eg. like the AddrBook (and its config https://github.com/tendermint/tendermint/blob/master/config/config.go#L212)